### PR TITLE
Example of DRY architecture leveraging new multiinclude feature

### PR DESCRIPTION
**Pull request by yorinasub17**
This PR updates this example repo to leverage the new multiinclude features to achieve a DRY-er architecture.

### CODE REVIEW

File: non-prod/us-east-1/qa/mysql/terragrunt.hcl [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
-  env = local.environment_vars.locals.environment
-}
-
-# Terragrunt will copy the Terraform configurations specified by the source parameter, along with any files in the
-# working directory, into a temporary folder, and execute your Terraform commands in that folder.
+# We override the terraform block source attribute here just for the QA environment to show how you would deploy a
+# different version of the module in a specific environment.
 terraform {
-  source = "git::git@github.com:gruntwork-io/terragrunt-infrastructure-modules-example.git//mysql?ref=v0.4.0"
+  source = "${include.envcommon.locals.base_source_url}?ref=v0.4.0"
```
Comment (brikis98) [inline non-prod/us-east-1/qa/mysql/terragrunt.hcl:10]: NIT: if you have the `terraform { source = "..." }` block in each child `terragrunt.hcl` anyway, is there a point in putting it into the `envcommon` files?

**Review by brikis98 — APPROVED** [review]
Very nice 👍 

It occurs to me that the multi `include` feature may warrant a new blog post at some point...

File: non-prod/us-east-1/qa/mysql/terragrunt.hcl (hunk 1 above)
Comment (yorinasub17) [reply non-prod/us-east-1/qa/mysql/terragrunt.hcl:10]: I only have this in the `qa` config. Note that it is missing from [prod](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example/blob/yori-dry-tg/prod/us-east-1/prod/mysql/terragrunt.hcl) and [stage](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example/blob/yori-dry-tg/non-prod/us-east-1/stage/mysql/terragrunt.hcl).

**Comment by yorinasub17** [comment]
Thanks for review! Merging this in now.

**Pull request merged**